storage/segment_query.go

```diff
@@ -562,6 +562,31 @@ func collectMultiFieldGroupByResults(
 	return groupByCalcIter.Err()
 }
 
+// NB: This method owns `maskingDocIDSetIt` and handles closing regardless of success or failure.
+func collectTimeBucketResults(
+	timestampField indexfield.TimeField,
+	maskingDocIDSetIter index.DocIDSetIterator,
+	res *query.TimeBucketResults,
+) error {
+	timestampIter, err := timestampField.Fetch(maskingDocIDSetIter)
+	if err != nil {
+		maskingDocIDSetIter.Close()
+		return fmt.Errorf("error fetching timestamp data: %v", err)
+	}
+	defer timestampIter.Close()
+
+	// TODO(xichen): This decodes the timestamp field again if the timestamp field is loaded from
```

Comment:
i'm not following this comment, i'll ask you offline.

Reply:
We're decoding once to apply filters to get the masking DocID iterator and again here when we collect the results.